Build failure on first run

Hiya,

When running a build on my Arch Linux machine I'm getting a build failure. The full build output is below:

```
which: no  in (/home/bardur/bin:/home/bardur/.cabal/bin:/usr/lib/ccache/bin:/usr/bin/vendor_perl:/home/bardur/.gem/ruby/1.8/bin:/home/bardur/.npm-installed/bin:/home/bardur/opt/purescript/.cabal-sandbox/bin:/home/bardur/opt/git-annex/bin:/home/bardur/bin:/home/bardur/.cabal/bin:/usr/lib/ccache/bin:/usr/bin/vendor_perl:/home/bardur/.gem/ruby/1.8/bin:/home/bardur/.npm-installed/bin:/home/bardur/opt/purescript/.cabal-sandbox/bin:/home/bardur/opt/git-annex/bin:/home/bardur/bin:/home/bardur/.cabal/bin:/usr/lib/ccache/bin:/usr/bin/vendor_perl:/home/bardur/.gem/ruby/1.8/bin:/home/bardur/.npm-installed/bin:/home/bardur/opt/purescript/.cabal-sandbox/bin:/home/bardur/opt/git-annex/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)
if [ ! -d build-24.5.1 ]; then mkdir build-24.5.1; fi
/usr/bin/emacs --eval "(add-to-list 'load-path (expand-file-name \"tests/compat\") 'append)" --eval "(when (< emacs-major-version 24) (setq byte-compile-warnings '(not cl-functions)))" --eval '(setq byte-compile-error-on-warn t)' --eval '(when (not (version< emacs-version "24.4")) (setq load-prefer-newer t))' --eval '(defun byte-compile-dest-file (filename) (concat (file-name-directory filename) "build-" emacs-version "/" (file-name-nondirectory filename) "c"))' --batch -Q -L . \
        --eval '(setq make-backup-files nil)' \
        --eval '(setq generated-autoload-file "/home/bardur/.Emacs/haskell-mode-13.20/haskell-mode-autoloads.el")' \
        -f batch-update-autoloads "."
LANG=en_US.UTF-8 makeinfo  -o haskell-mode.info doc/haskell-mode.texi
/usr/bin/emacs --eval "(add-to-list 'load-path (expand-file-name \"tests/compat\") 'append)" --eval "(when (< emacs-major-version 24) (setq byte-compile-warnings '(not cl-functions)))" --eval '(setq byte-compile-error-on-warn t)' --eval '(when (not (version< emacs-version "24.4")) (setq load-prefer-newer t))' --eval '(defun byte-compile-dest-file (filename) (concat (file-name-directory filename) "build-" emacs-version "/" (file-name-nondirectory filename) "c"))' --batch -Q -L . -f batch-byte-compile-if-not-done haskell-sandbox.el haskell-modules.el haskell-complete-module.el haskell-checkers.el haskell-session.el inf-haskell.el highlight-uses-mode.el haskell-process.el haskell-collapse.el haskell-hoogle.el haskell-presentation-mode.el haskell-cabal.el haskell-font-lock.el haskell-move-nested.el haskell-mode.el haskell-load.el ghci-script-mode.el ghc-core.el haskell-unicode-input-method.el w3m-haddock.el haskell-string.el haskell-debug.el haskell-navigate-imports.el haskell-menu.el haskell-interactive-mode.el haskell.el haskell-repl.el haskell-compile.el haskell-utils.el haskell-completions.el haskell-doc.el haskell-compat.el haskell-align-imports.el haskell-commands.el haskell-indent.el haskell-decl-scan.el haskell-customize.el haskell-indentation.el haskell-c2hs.el haskell-sort-imports.el haskell-lexeme.el
Wrote /home/bardur/.Emacs/haskell-mode-13.20/haskell-mode-autoloads.el
Generating autoloads for ghc-core.el...
Generating autoloads for ghc-core.el...done
Generating autoloads for ghci-script-mode.el...
Generating autoloads for haskell-interactive-mode.el...
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-sandbox.elc
Generating autoloads for haskell-interactive-mode.el...done
Generating autoloads for haskell-lexeme.el...
Generating autoloads for haskell-mode-pkg.el...
Generating autoloads for haskell-mode-pkg.el...done
Generating autoloads for haskell-mode.el...
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-modules.elc
Generating autoloads for haskell-mode.el...done
Generating autoloads for haskell-modules.el...
Generating autoloads for haskell-modules.el...done
Generating autoloads for haskell-utils.el...done
Generating autoloads for haskell.el...
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-complete-module.elc
Generating autoloads for haskell.el...done
Generating autoloads for highlight-uses-mode.el...
Generating autoloads for highlight-uses-mode.el...done
Generating autoloads for inf-haskell.el...
Generating autoloads for inf-haskell.el...done
Generating autoloads for w3m-haddock.el...
Generating autoloads for w3m-haddock.el...done
Saving file /home/bardur/.Emacs/haskell-mode-13.20/haskell-mode-autoloads.el...
Wrote /home/bardur/.Emacs/haskell-mode-13.20/haskell-mode-autoloads.el
(No changes need to be saved)
# check if autoloads will really load
/usr/bin/emacs --eval "(add-to-list 'load-path (expand-file-name \"tests/compat\") 'append)" --eval "(when (< emacs-major-version 24) (setq byte-compile-warnings '(not cl-functions)))" --eval '(setq byte-compile-error-on-warn t)' --eval '(when (not (version< emacs-version "24.4")) (setq load-prefer-newer t))' --eval '(defun byte-compile-dest-file (filename) (concat (file-name-directory filename) "build-" emacs-version "/" (file-name-nondirectory filename) "c"))' --batch -Q -L . -l "haskell-mode-autoloads.el"
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-checkers.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-session.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/inf-haskell.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/highlight-uses-mode.elc

In toplevel form:
haskell-process.el:488:1:Warning: Unused lexical argument `e'
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-process.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-cabal.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-font-lock.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-move-nested.elc

In toplevel form:
haskell-mode.el:1168:1:Error: the function `haskell-completions-completion-at-point' is not known to be defined.

In toplevel form:
haskell-load.el:93:1:Warning: Unused lexical argument `e'
install-info --dir=dir haskell-mode.info
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-load.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/ghci-script-mode.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-sort-imports.elc
Wrote /home/bardur/.Emacs/haskell-mode-13.20/build-24.5.1/haskell-lexeme.elc
Makefile:66: recipe for target 'build-24.5.1' failed
make: *** [build-24.5.1] Error 1
```
(I've elided quite a few "Wrote ..." and "Generated autoloads..." lines just for brevity.)

If I re-run make I get

```
which: no  in (/home/... [here it lists out my full  $PATH])
```

and no error.

I *think* everything works as it should, but it's a bit disconcerting to see the initial error.

--- a/haskell-completions.el
+++ b/haskell-completions.el
@@ -301,7 +301,7 @@ PREFIX should be a list such one returned by
                         haskell-completions--keywords)))))
         (list beg end candidates)))))
 
-
+;;;###autoload
 (defun haskell-completions-completion-at-point ()
   "Provide completion list for thing at point.
 This function is used in non-interactive `haskell-mode'.  It

--- a/haskell-mode.el
+++ b/haskell-mode.el
@@ -838,7 +838,7 @@ Minor modes that work well with `haskell-mode':
   (add-hook 'after-save-hook 'haskell-mode-after-save-handler nil t)
   ;; provide non-interactive completion function
   (add-hook 'completion-at-point-functions
-            #'haskell-completions-completion-at-point
+            'haskell-completions-completion-at-point
             nil
             t)
   (haskell-indentation-mode))

